src/frontend/details/Details.js

```diff
@@ -25,42 +24,47 @@ export default class StatefulDetails extends Component {
       (result) => {
         this.setState({
           isLoaded: true,
-          shows: result
+          data: result
         });
       },
-      // Note: it's important to handle errors here
-      // instead of a catch() block so that we don't swallow
-      // exceptions from actual bugs in components.
       (error) => {
         this.setState({
           isLoaded: true,
-          error
+          error: error
         });
       }
     )
-
-
+    .then( () => {
     let found = this.state.data.find((movie) => movie.id === this.props.match.params.id)
     let name = found ? found.title : <Redirect to='/not-found' />
```

Comment:
I find this a bit confusing.

IMHO, you should:

1) Display the loading message by default (e.g. the list of movies is empty, and your  `notFound` flag is not set
2) Get the data from the backend
3) Find the show you want to display
4) Save it into the state if you found it, or set the `notFound` flag to true otherwise
5) in the `render` function, return `<Redirect...` if the `notFound` flag is set, and the normal markup for `Details` otherwise